src/install/lifecycle_script_runner.zig

```diff
@@ -132,13 +132,40 @@ pub const LifecycleScriptSubprocess = struct {
 
         const combined_script: [:0]u8 = copy_script.items[0 .. copy_script.items.len - 1 :0];
 
+        var path_buf: [bun.MAX_PATH_BYTES]u8 = undefined;
+        var path_buf2: [bun.MAX_PATH_BYTES]u8 = undefined;
+        const lifecycle_script_path, var lifecycle_script_file = if (Environment.isWindows or shell_bin == null) blk: {
+            const tempdir = this.manager.getTemporaryDirectory();
+            var lifecycle_script_name = bun.span(try bun.fs.FileSystem.instance.tmpname("lifecycle-script", &path_buf, 12345));
```

Comment:
why are we saving files to a temporary directory? there are many alternatives which would be more reliable

Reply:
it creates a temporary file (a `.bun.sh`) in the global temp directory and then runs it with the bun shell in a child process

Reply:
why not pass the script contents as argv? or pass it as stdin? or pass it as an environment variable? any of these would be more reliable + low overhead compared to writing to a temporary file

Reply:
im unaware of any way to use bun's shell interpreter via standard in or via argv. `--eval` and `--print` apply to typescript only. so a mechanism would have to be added for this (`--eval-shell`? i dont want to add more flags than needed)